ui/src/components/Timeline/Timeline.tsx

```diff
@@ -131,6 +127,9 @@ const Timeline: FC<TimelineProps> = ({
             writeable={writeable}
             onVertexChange={(vertex: Vertex) => {
               dispatch({ type: 'UPDATE_VERTEX', payload: { vertex } });
+              // State updates are executed asyncronously, so we need to compute the new layout explicitly
+              const newLayout = updateVertex(state.layout, vertex);
+              onLayoutUpdate(newLayout);
```

Comment:
We now call the callback explicitly after the user has layout-mutating actions (right now, the only such action is changing the color of an entity).